### ISSUE TITLE
The custom integration 'tesla_cc' does not have a valid version key (None) in the manifest file and was blocked from loading

### ISSUE THREAD

**Issue by purcell-lab** [mlowijs/hass_tesla_v2#7]
**Describe the bug**

homeassistant.log
ERROR (SyncWorker_1) [homeassistant.loader] The custom integration 'tesla_cc' does not have a valid version key (None) in the manifest file and was blocked from loading. See https://developers.home-assistant.io/blog/2021/01/29/custom-integration-changes#versions for more details

**To Reproduce**
Everytime startup

**Expected behavior**

Valid Version FIle to be included:
See https://developers.home-assistant.io/blog/2021/01/29/custom-integration-changes#versions for more details

**Comment by stefanstranger** [comment]
You can solve this yourself by adding a version property to the manifest.json in folder /config/custom_components/tesla_cc

```json
{
  "domain": "tesla_cc",
  "name": "Tesla Custom Component",
  "version": "0.0.1",
  "documentation": "https://github.com/mlowijs/hass_tesla_v2",
  "dependencies": [],
  "codeowners": ["@mlowijs", "@robhofmann"],
  "requirements": []
}
```

HTH
Stefan

**Comment by purcell-lab** [comment]
## Status update — May 2026

**Repo state**

For anyone landing here in 2026: this repository has been **effectively abandoned**. The last commit was June 2020. The original issue was about the `version` key being required in `manifest.json` starting with Home Assistant 2021.x — and the fix is still the one @stefanstranger posted in August 2021.

**Workaround** (still valid)

Manually add `"version": "0.0.1"` (or any valid semver) to `/config/custom_components/tesla_cc/manifest.json`:

```json
{
  "domain": "tesla_cc",
  "name": "Tesla Custom Component",
  "version": "0.0.1",
  "documentation": "https://github.com/mlowijs/hass_tesla_v2",
  "dependencies": [],
  "codeowners": ["@mlowijs", "@robhofmann"],
  "requirements": []
}
```

**Recommended way forward**

This component is far behind the actively-maintained alternatives. For new installs / migrations, please use:

- **[`alandtse/tesla`](https://github.com/alandtse/tesla)** (Tesla Custom Integration) — currently on `v3.26.0` (May 2026), active development, supports vehicles + Powerwall + solar via the Tesla Fleet API.
- **[`Teslemetry/hass-teslemetry`](https://github.com/Teslemetry/hass-teslemetry)** — subscription-backed Teslemetry service, active development.
- Native **Home Assistant Tesla integration** for vehicle-only setups.

This issue can probably be closed as **wontfix / abandoned** — the workaround unblocks the immediate startup error, and continuing to run this component on current HA isn't recommended. I'll defer to @mlowijs on the official disposition.
